### ISSUE TITLE
1.5.3: pytest is failing

### ISSUE THREAD

**Issue by kloczek** [getlogbook/logbook#318]
I'm trying to package your module as rpm packag. So I'm using typical in such case build, install and test cycle used on building package from non-root account:
- "setup.py build"
- "setup.py install --root </install/prefix>"
- "pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

May I ask for help because few units are failing:
```console
+ /usr/bin/pytest -ra
1549013:C 14 Aug 2021 09:10:05.214 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1549013:C 14 Aug 2021 09:10:05.214 # Redis version=6.2.2, bits=64, commit=00000000, modified=0, pid=1549013, just started
1549013:C 14 Aug 2021 09:10:05.215 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1549013:M 14 Aug 2021 09:10:05.216 * Increased maximum number of open files to 10032 (it was originally set to 1024).
1549013:M 14 Aug 2021 09:10:05.216 * monotonic clock: POSIX clock_gettime
                _._
           _.-``__ ''-._
      _.-``    `.  `_.  ''-._           Redis 6.2.2 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 1549013
  `-._    `-._  `-./  _.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |           https://redis.io
  `-._    `-._`-.__.-'_.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |
  `-._    `-._`-.__.-'_.-'    _.-'
      `-._    `-.__.-'    _.-'
          `-._        _.-'
              `-.__.-'

1549013:M 14 Aug 2021 09:10:05.217 # Server initialized
1549013:M 14 Aug 2021 09:10:05.217 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1549013:M 14 Aug 2021 09:10:05.219 * Ready to accept connections
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.11, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/tkloczko/rpmbuild/BUILD/logbook-1.5.3
plugins: forked-1.3.0, shutil-1.7.0, virtualenv-1.7.0, expect-1.1.0, flake8-1.0.7, timeout-1.4.2, betamax-0.8.1, freezegun-0.4.2, case-1.5.3, aspectlib-1.5.2, toolbox-0.5, rerunfailures-9.1.1, requests-mock-1.9.3, cov-2.12.1, pyfakefs-4.5.0, flaky-3.7.0, benchmark-3.4.1, xdist-2.3.0, pylama-7.7.1, datadir-1.3.1, regressions-2.2.0, cases-3.6.3, xprocess-0.18.1, black-0.3.12, checkdocs-2.7.1, anyio-3.3.0, Faker-8.11.0, asyncio-0.15.1, trio-0.7.0, httpbin-1.0.0, subtests-0.5.0, isort-2.0.0, hypothesis-6.14.6, mock-3.6.1, profiling-1.7.0
collected 428 items

. .                                                                                                                                                                  [  0%]
tests/test_asyncio.py .F                                                                                                                                             [  0%]
tests/test_ci.py ssss                                                                                                                                                [  1%]
tests/test_deadlock.py ..                                                                                                                                            [  2%]
tests/test_file_handler.py ................................................................                                                                          [ 17%]
tests/test_fingers_crossed_handler.py ............                                                                                                                   [ 19%]
tests/test_flags.py ....                                                                                                                                             [ 20%]
tests/test_groups.py ....                                                                                                                                            [ 21%]
tests/test_handler_errors.py ......                                                                                                                                  [ 23%]
tests/test_handlers.py ......................                                                                                                                        [ 28%]
tests/test_helpers.py ....                                                                                                                                           [ 29%]
tests/test_log_record.py ....................                                                                                                                        [ 33%]
tests/test_logbook.py ......                                                                                                                                         [ 35%]
tests/test_logger.py ......                                                                                                                                          [ 36%]
tests/test_logging_api.py ..........................                                                                                                                 [ 42%]
tests/test_logging_compat.py ........                                                                                                                                [ 44%]
tests/test_logging_times.py ..........                                                                                                                               [ 47%]
tests/test_mail_handler.py ..............                                                                                                                            [ 50%]
tests/test_more.py .........sss.........sss                                                                                                                          [ 55%]
tests/test_nteventlog_handler.py ss                                                                                                                                  [ 56%]
tests/test_null_handler.py ............                                                                                                                              [ 59%]
tests/test_processors.py ........                                                                                                                                    [ 61%]
tests/test_queues.py ............................                                                                                                                    [ 67%]
tests/test_syslog_handler.py ........................                                                                                                                [ 73%]
tests/test_test_handler.py ............................................................                                                                              [ 87%]
tests/test_ticketing.py FF                                                                                                                                           [ 87%]
tests/test_unicode.py ..............                                                                                                                                 [ 91%]
tests/test_utils.py ......................................                                                                                                           [100%]

================================================================================= FAILURES =================================================================================
__________________________________________________________________ test_asyncio_context_management[True] ___________________________________________________________________

logger = <logbook.base.Logger object at 0x7fc58ec66340>

    @pytest.mark.skipif(not has_contextvars, reason="Contexvars not available")
    def test_asyncio_context_management(logger):
        h1 = logbook.TestHandler()
        h2 = logbook.TestHandler()

        async def task(handler, msg):
            for _ in range(ITERATIONS):
                with handler.contextbound():
                    logger.info(msg)

                await asyncio.sleep(0)  # allow for context switch

        asyncio.get_event_loop().run_until_complete(asyncio.gather(task(h1, 'task1'), task(h2, 'task2')))

>       assert len(h1.records) == ITERATIONS
E       assert 200 == 100
E        +  where 200 = len([<logbook.base.LogRecord object at 0x7fc58eb443a0>, <logbook.base.LogRecord object at 0x7fc58eb44430>, <logbook.base.L...c58eb445b0>, <logbook.base.LogRecord object at 0x7fc58eb44640>, <logbook.base.LogRecord object at 0x7fc58eb446d0>, ...])
E        +    where [<logbook.base.LogRecord object at 0x7fc58eb443a0>, <logbook.base.LogRecord object at 0x7fc58eb44430>, <logbook.base.L...c58eb445b0>, <logbook.base.LogRecord object at 0x7fc58eb44640>, <logbook.base.LogRecord object at 0x7fc58eb446d0>, ...] = <logbook.handlers.TestHandler object at 0x7fc58ec663d0>.records

tests/test_asyncio.py:23: AssertionError
_______________________________________________________________________ test_basic_ticketing[False] ________________________________________________________________________

logger = <logbook.base.Logger object at 0x7fc58e8f29a0>

    @pytest.mark.xfail(
        os.name == 'nt' and (python_version == (3, 2) or python_version == (3, 3)),
        reason='Problem with in-memory sqlite on Python 3.2, 3.3 and Windows')
    @require_module('sqlalchemy')
    def test_basic_ticketing(logger):
        from logbook.ticketing import TicketingHandler
        from time import sleep
        with TicketingHandler('sqlite:///') as handler:
            for x in xrange(5):
                logger.warn('A warning')
                sleep(0.2)
                logger.info('An error')
                sleep(0.2)
                if x < 2:
                    try:
                        1 / 0
                    except Exception:
                        logger.exception()

>       assert handler.db.count_tickets() == 3

tests/test_ticketing.py:41:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <logbook.ticketing.SQLAlchemyBackend object at 0x7fc58e8f2550>

    def count_tickets(self):
        """Returns the number of tickets."""
>       return self.engine.execute(self.tickets.count()).fetchone()[0]
E       AttributeError: 'Table' object has no attribute 'count'

logbook/ticketing.py:246: AttributeError
________________________________________________________________________ test_basic_ticketing[True] ________________________________________________________________________

logger = <logbook.base.Logger object at 0x7fc5ba958430>

    @pytest.mark.xfail(
        os.name == 'nt' and (python_version == (3, 2) or python_version == (3, 3)),
        reason='Problem with in-memory sqlite on Python 3.2, 3.3 and Windows')
    @require_module('sqlalchemy')
    def test_basic_ticketing(logger):
        from logbook.ticketing import TicketingHandler
        from time import sleep
        with TicketingHandler('sqlite:///') as handler:
            for x in xrange(5):
                logger.warn('A warning')
                sleep(0.2)
                logger.info('An error')
                sleep(0.2)
                if x < 2:
                    try:
                        1 / 0
                    except Exception:
                        logger.exception()

>       assert handler.db.count_tickets() == 3

tests/test_ticketing.py:41:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <logbook.ticketing.SQLAlchemyBackend object at 0x7fc5ba958490>

    def count_tickets(self):
        """Returns the number of tickets."""
>       return self.engine.execute(self.tickets.count()).fetchone()[0]
E       AttributeError: 'Table' object has no attribute 'count'

logbook/ticketing.py:246: AttributeError
============================================================================= warnings summary =============================================================================
tests/test_queues.py::test_missing_zeromq[False]
tests/test_queues.py::test_missing_zeromq[True]
  /usr/lib/python3.8/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function ZeroMQHandler.__del__ at 0x7fc58e9189d0>

  Traceback (most recent call last):
    File "/home/tkloczko/rpmbuild/BUILD/logbook-1.5.3/logbook/queues.py", line 241, in __del__
      self.close(linger=100)
    File "/home/tkloczko/rpmbuild/BUILD/logbook-1.5.3/logbook/queues.py", line 232, in close
      self.socket.close(linger)
  AttributeError: 'ZeroMQHandler' object has no attribute 'socket'

    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

tests/test_ticketing.py::test_basic_ticketing[False]
tests/test_ticketing.py::test_basic_ticketing[True]
  /home/tkloczko/rpmbuild/BUILD/logbook-1.5.3/logbook/ticketing.py:147: SADeprecationWarning: The create_engine.convert_unicode parameter and corresponding dialect-level parameters are deprecated, and will be removed in a future release.  Modern DBAPIs support Python Unicode natively and this parameter is unnecessary.
    self.engine = create_engine(engine_or_uri, convert_unicode=True,

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [2] tests/test_ci.py:8: AppVeyor CI test
SKIPPED [2] tests/test_ci.py:16: Travis CI test
SKIPPED [2] tests/test_more.py:166: Module riemann_client is required
SKIPPED [2] tests/test_more.py:189: Module riemann_client is required
SKIPPED [2] tests/test_more.py:195: Module riemann_client is required
SKIPPED [2] tests/test_nteventlog_handler.py:9: Don't clutter NT Event Log unless enabled.
FAILED tests/test_asyncio.py::test_asyncio_context_management[True] - assert 200 == 100
FAILED tests/test_ticketing.py::test_basic_ticketing[False] - AttributeError: 'Table' object has no attribute 'count'
FAILED tests/test_ticketing.py::test_basic_ticketing[True] - AttributeError: 'Table' object has no attribute 'count'
========================================================== 3 failed, 412 passed, 12 skipped, 4 warnings in 33.27s ==========================================================
pytest-xprocess reminder::Be sure to terminate the started process by running 'pytest --xkill' if you have not explicitly done so in your fixture with 'xprocess.getinfo(<process_name>).terminate()'.
```

**Comment by kloczek** [comment]
pytest seems still is failing but now I have onlty one failing unit
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-logbook-1.5.3-9.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-logbook-1.5.3-9.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
824545:C 13 Jan 2022 14:00:57.469 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
824545:C 13 Jan 2022 14:00:57.469 # Redis version=6.2.6, bits=64, commit=00000000, modified=0, pid=824545, just started
824545:C 13 Jan 2022 14:00:57.469 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
824545:M 13 Jan 2022 14:00:57.470 * Increased maximum number of open files to 10032 (it was originally set to 1024).
824545:M 13 Jan 2022 14:00:57.470 * monotonic clock: POSIX clock_gettime
                _._
           _.-``__ ''-._
      _.-``    `.  `_.  ''-._           Redis 6.2.6 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 824545
  `-._    `-._  `-./  _.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |           https://redis.io
  `-._    `-._`-.__.-'_.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |
  `-._    `-._`-.__.-'_.-'    _.-'
      `-._    `-.__.-'    _.-'
          `-._        _.-'
              `-.__.-'

824545:M 13 Jan 2022 14:00:57.471 # Server initialized
824545:M 13 Jan 2022 14:00:57.472 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
824545:M 13 Jan 2022 14:00:57.473 * Ready to accept connections
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/logbook-1.5.3
collected 213 items

tests/test_asyncio.py .                                                                                                                                              [  0%]
tests/test_ci.py ss                                                                                                                                                  [  1%]
tests/test_deadlock.py .                                                                                                                                             [  1%]
tests/test_file_handler.py ................................                                                                                                          [ 16%]
tests/test_fingers_crossed_handler.py ......                                                                                                                         [ 19%]
tests/test_flags.py ..                                                                                                                                               [ 20%]
tests/test_groups.py ..                                                                                                                                              [ 21%]
tests/test_handler_errors.py ...                                                                                                                                     [ 23%]
tests/test_handlers.py ...........                                                                                                                                   [ 28%]
tests/test_helpers.py ..                                                                                                                                             [ 29%]
tests/test_log_record.py ..........                                                                                                                                  [ 33%]
tests/test_logbook.py ...                                                                                                                                            [ 35%]
tests/test_logger.py ...                                                                                                                                             [ 36%]
tests/test_logging_api.py .............                                                                                                                              [ 42%]
tests/test_logging_compat.py ....                                                                                                                                    [ 44%]
tests/test_logging_times.py ss...                                                                                                                                    [ 46%]
tests/test_mail_handler.py .......                                                                                                                                   [ 50%]
tests/test_more.py .........sss                                                                                                                                      [ 55%]
tests/test_nteventlog_handler.py s                                                                                                                                   [ 56%]
tests/test_null_handler.py ......                                                                                                                                    [ 59%]
tests/test_processors.py ....                                                                                                                                        [ 61%]
tests/test_queues.py .........s....                                                                                                                                  [ 67%]
tests/test_syslog_handler.py ............                                                                                                                            [ 73%]
tests/test_test_handler.py ..............................                                                                                                            [ 87%]
tests/test_ticketing.py F                                                                                                                                            [ 87%]
tests/test_unicode.py .......                                                                                                                                        [ 91%]
tests/test_utils.py ...................                                                                                                                              [100%]

================================================================================= FAILURES =================================================================================
___________________________________________________________________________ test_basic_ticketing ___________________________________________________________________________

logger = <logbook.base.Logger object at 0x7f75f9821a00>

    @pytest.mark.xfail(
        os.name == 'nt' and (python_version == (3, 2) or python_version == (3, 3)),
        reason='Problem with in-memory sqlite on Python 3.2, 3.3 and Windows')
    @require_module('sqlalchemy')
    def test_basic_ticketing(logger):
        from logbook.ticketing import TicketingHandler
        from time import sleep
        with TicketingHandler('sqlite:///') as handler:
            for x in xrange(5):
                logger.warn('A warning')
                sleep(0.2)
                logger.info('An error')
                sleep(0.2)
                if x < 2:
                    try:
                        1 / 0
                    except Exception:
                        logger.exception()

>       assert handler.db.count_tickets() == 3

tests/test_ticketing.py:41:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <logbook.ticketing.SQLAlchemyBackend object at 0x7f75f98211c0>

    def count_tickets(self):
        """Returns the number of tickets."""
>       return self.engine.execute(self.tickets.count()).fetchone()[0]
E       AttributeError: 'Table' object has no attribute 'count'

logbook/ticketing.py:246: AttributeError
============================================================================= warnings summary =============================================================================
tests/test_queues.py::test_missing_zeromq
  /usr/lib/python3.8/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function ZeroMQHandler.__del__ at 0x7f75f9771940>

  Traceback (most recent call last):
    File "/home/tkloczko/rpmbuild/BUILD/logbook-1.5.3/logbook/queues.py", line 241, in __del__
      self.close(linger=100)
    File "/home/tkloczko/rpmbuild/BUILD/logbook-1.5.3/logbook/queues.py", line 232, in close
      self.socket.close(linger)
  AttributeError: 'ZeroMQHandler' object has no attribute 'socket'

    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

tests/test_ticketing.py::test_basic_ticketing
  /home/tkloczko/rpmbuild/BUILD/logbook-1.5.3/logbook/ticketing.py:147: SADeprecationWarning: The create_engine.convert_unicode parameter and corresponding dialect-level parameters are deprecated, and will be removed in a future release.  Modern DBAPIs support Python Unicode natively and this parameter is unnecessary.
    self.engine = create_engine(engine_or_uri, convert_unicode=True,

-- Docs: https://docs.pytest.org/en/stable/warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [1] tests/test_ci.py:8: AppVeyor CI test
SKIPPED [1] tests/test_ci.py:16: Travis CI test
SKIPPED [2] tests/test_logging_times.py:35: Cannot test utc/localtime differences if they vary by less than one minute...
SKIPPED [1] tests/test_more.py:166: Module riemann_client is required
SKIPPED [1] tests/test_more.py:189: Module riemann_client is required
SKIPPED [1] tests/test_more.py:195: Module riemann_client is required
SKIPPED [1] tests/test_nteventlog_handler.py:9: Don't clutter NT Event Log unless enabled.
SKIPPED [1] tests/test_queues.py:166: Module execnet is required
FAILED tests/test_ticketing.py::test_basic_ticketing - AttributeError: 'Table' object has no attribute 'count'
========================================================== 1 failed, 203 passed, 9 skipped, 2 warnings in 13.95s ===========================================================
```

**Comment by RazerM** [comment]
Resolved by #331
